docs/ldml/tr35-personNames.md

```diff
@@ -763,29 +762,33 @@ There are two main challenges in dealing with foreign name formatting that needs
 * “[미야자키<span style="background-color:aqua">　</span>하야오(일본어: 宮﨑<span style="background-color:aqua">　</span>駿 Miyazaki Hayao](https://ko.wikipedia.org/wiki/%EB%AF%B8%EC%95%BC%EC%9E%90%ED%82%A4_%ED%95%98%EC%95%BC%EC%98%A4)…”. 
 
-2. If a locale **doesn’t** require spaces between words, there are two cases, based on whether the name is foreign or not (based on the PersonName objects explicit or calculated locale's language subtag). For example, the formatting locale might be Japanese, and the locale of the PersonName object might be de_CH, German (Switzerland), such as Albert Einstein.
+If a locale **doesn’t** require spaces between words, there are two cases, based on whether the name is foreign or not (based on the PersonName objects explicit or calculated locale's language subtag). For example, the formatting locale might be Japanese, and the locale of the PersonName object might be de_CH, German (Switzerland), such as Albert Einstein. When the locale is foreign, the **foreignSpaceReplacement** is substituted for each space in the formatted name. When the name locale is native, a **nativeSpaceReplacement** is substituted for each space in the formatted name. The precise algorithm is given below.
 
-    1. **The name is native** In that case, the patterns are used as is.
-    2. **The name is foreign** In that case, the **foreignSpaceReplacement** is substituted for each space formatted name. Here are examples for Albert Einstein in Japanese and Chinese:
+Here are examples for Albert Einstein in Japanese and Chinese:
         * [アルベルト<span style="background-color:aqua">・</span>アインシュタイン](https://ja.wikipedia.org/wiki/%E3%82%A2%E3%83%AB%E3%83%99%E3%83%AB%E3%83%88%E3%83%BB%E3%82%A2%E3%82%A4%E3%83%B3%E3%82%B7%E3%83%A5%E3%82%BF%E3%82%A4%E3%83%B3) 
         * [阿尔伯特<span style="background-color:aqua">·</span>爱因斯坦](https://zh.wikipedia.org/wiki/%E9%98%BF%E5%B0%94%E4%BC%AF%E7%89%B9%C2%B7%E7%88%B1%E5%9B%A0%E6%96%AF%E5%9D%A6) 
 
+#### 6.8.1 Setting the spaceReplacement
+
+1. The foreignSpaceReplacement is provided by the value for the `foreignSpaceReplacement` element; the default value is " ".
+
+2. The nativeSpaceReplacement is determined by the following algorithm, chosing between " " and "".
+
+1. Get the script of the formatting locale
+2. If the likely script is Thai, let nativeSpaceReplacement = " "
+3. Otherwise let nativeSpaceReplacement = "" if either of the following applies:
+    1. The script is Jpan, Hant, or Hans
+    2. The script has the script metadata property lbLetters = YES (this can also be algorithmically derived from the LineBreak property data).
+5. Otherwise, let nativeSpaceReplacement = ""
+
+3. If the language subtag of the formatter's locale equals the language subtag of the name's locale, then let spaceReplacement = nativeSpaceReplacement, otherwise let spaceReplacement = foreignSpaceReplacement.
```

Comment:
That's not what I implemented, and I'm not sure it's what we want.  See my comments in my PR.

Reply:
Okay, I've gone back and updated https://github.com/unicode-org/icu/pull/2356 based on our discussion today.  I think it's a lot simpler.  I changed a bunch of stuff in my unit tests to match the new logic, but I think the changes make sense, and everything in the `ExhaustivePersonNameFormatterTest` that passed before still passes.

The one thing I'm not happy with is the logic to decide whether to use the foreign or native space replacement character.  I decided you were right that looking for common scripts between the locales was a bad idea, so I'm just looking at the language, but then I had to put a hack in there to get Japanese and Chinese to format correctly in each other's locales, and I really don't like that hack.  If you've got a smarter idea, I'm all ears.

Reply:
I kept the hack in the spec; for a data-driven solution we'll have to wait for v44